core/src/test/java/org/elasticsearch/index/query/MoreLikeThisQueryBuilderTests.java

```diff
@@ -0,0 +1,309 @@
+        } else {
+            item.doc(randomArtificialDoc());
+        }
+        // per field analyzer
+        if (randomBoolean()) {
+            item.perFieldAnalyzer(randomPerFieldAnalyzer());
+        }
+        return item;
+    }
+
+    private XContentBuilder randomArtificialDoc() {
+        XContentBuilder doc;
+        try {
+            doc = jsonBuilder().startObject();
+            for (String field : randomFields) {
+                doc.field(field, randomAsciiOfLength(10));
+            }
+        } catch (IOException e) {
+            throw new ElasticsearchException("Unable to generate random artificial doc!");
+        }
+        return doc;
+    }
+
+    private Map<String, String> randomPerFieldAnalyzer() {
+        Map<String, String> perFieldAnalyzer = new HashMap<>();
+        for (String field : randomFields) {
+            perFieldAnalyzer.put(field, randomAnalyzer());
+        }
+        return perFieldAnalyzer;
+    }
+
+    @Override
+    protected MoreLikeThisQueryBuilder doCreateTestQueryBuilder() {
+        MoreLikeThisQueryBuilder queryBuilder;
+        if (randomBoolean()) { // for the default field
+            queryBuilder = new MoreLikeThisQueryBuilder();
+        } else {
+            queryBuilder = new MoreLikeThisQueryBuilder(randomFields);
+        }
+        // like field is required
+        if (randomBoolean()) {
+            queryBuilder.like(generateRandomStringArray(5, 5, false, false));
+        } else {
+            queryBuilder.like(randomLikeItems);
+        }
+        if (randomBoolean()) {
+            queryBuilder.unlike(generateRandomStringArray(5, 5, false, false));
+        }
+        if (randomBoolean()) {
+            queryBuilder.unlike(randomUnlikeItems);
+        }
+        if (randomBoolean()) {
+            queryBuilder.maxQueryTerms(randomInt(25));
+        }
+        if (randomBoolean()) {
+            queryBuilder.minTermFreq(randomInt(5));
+        }
+        if (randomBoolean()) {
+            queryBuilder.minDocFreq(randomInt(5));
+        }
+        if (randomBoolean()) {
+            queryBuilder.maxDocFreq(randomInt(100));
+        }
+        if (randomBoolean()) {
+            queryBuilder.minWordLength(randomInt(5));
+        }
+        if (randomBoolean()) {
+            queryBuilder.maxWordLength(randomInt(25));
+        }
+        if (randomBoolean()) {
+            queryBuilder.stopWords(generateRandomStringArray(5, 5, false));
+        }
+        if (randomBoolean()) {
+            queryBuilder.analyzer(randomAnalyzer());  // fix the analyzer?
+        }
+        if (randomBoolean()) {
+            queryBuilder.minimumShouldMatch(randomMinimumShouldMatch());
+        }
+        if (randomBoolean()) {
+            queryBuilder.boostTerms(randomFloat() * 10);
+        }
+        if (randomBoolean()) {
+            queryBuilder.include(true);
+        }
+        if (randomBoolean()) {
+            queryBuilder.failOnUnsupportedField(false);
```

Comment:
OK sounds good, however we do now have 2/3 of chance to get true, but I understand that in general we should be testing unset.